### PR TITLE
[fix][flask-test] Fix and improve LookupRetryTest

### LINKED ISSUE #17785: Flaky-test: LookupRetryTest.testCloseConnectionOnBrokerRejectedRequest
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failure

https://github.com/apache/pulsar/actions/runs/3101892132/jobs/5023814359

### Exception stacktrace

```
Error:  Tests run: 37, Failures: 1, Errors: 0, Skipped: 36, Time elapsed: 8.833 s <<< FAILURE! - in org.apache.pulsar.client.impl.LookupRetryTest
  Error:  testCloseConnectionOnBrokerRejectedRequest(org.apache.pulsar.client.impl.LookupRetryTest)  Time elapsed: 2.146 s  <<< FAILURE!
  java.lang.AssertionError: expected [2] but found [1]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:907)
  	at org.testng.Assert.assertEquals(Assert.java:917)
  	at org.apache.pulsar.client.impl.LookupRetryTest.testCloseConnectionOnBrokerRejectedRequest(LookupRetryTest.java:217)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by coderzc**
Fixes #17785

### Motivation

The `failureMap` need to be clear after run per unit test.

### Modifications

Clear `failureMap` after run per unit test, and only run once `setup()`/`cleanup()` to reduce execution time.

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### Matching PR in forked repository

PR in forked repository: https://github.com/coderzc/pulsar/pull/6

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
LGTM. Nice catch!

**Comment by tisonkun** [comment]
@coderzc although, can you explain a bit about how this patch fix the original issue? 

The original failure reports:

```
  Error:  testCloseConnectionOnBrokerRejectedRequest(org.apache.pulsar.client.impl.LookupRetryTest)  Time elapsed: 2.146 s  <<< FAILURE!
  java.lang.AssertionError: expected [2] but found [1]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:907)
  	at org.testng.Assert.assertEquals(Assert.java:917)
  	at org.apache.pulsar.client.impl.LookupRetryTest.testCloseConnectionOnBrokerRejectedRequest(LookupRetryTest.java:217)
```

That doesn't directly related to the hash map.

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/LookupRetryTest.java [hunk 1]
```diff
@@ -94,12 +88,18 @@ protected ServerCnx newServerCnx(PulsarService pulsar, String listenerName) thro
         };
     }
 
-    @AfterMethod(alwaysRun = true)
+    @AfterClass
```
Comment (tisonkun) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/impl/LookupRetryTest.java:91]:
```suggestion
    @AfterClass(alwaysRun = true)
```
?

**Comment by coderzc** [comment]
> @coderzc although, can you explain a bit about how this patch fix the original issue?
> 
> The original failure reports:
> 
> ```
>   Error:  testCloseConnectionOnBrokerRejectedRequest(org.apache.pulsar.client.impl.LookupRetryTest)  Time elapsed: 2.146 s  <<< FAILURE!
>   java.lang.AssertionError: expected [2] but found [1]
>   	at org.testng.Assert.fail(Assert.java:99)
>   	at org.testng.Assert.failNotEquals(Assert.java:1037)
>   	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
>   	at org.testng.Assert.assertEquals(Assert.java:122)
>   	at org.testng.Assert.assertEquals(Assert.java:907)
>   	at org.testng.Assert.assertEquals(Assert.java:917)
>   	at org.apache.pulsar.client.impl.LookupRetryTest.testCloseConnectionOnBrokerRejectedRequest(LookupRetryTest.java:217)
> ```
> 
> That doesn't directly related to the hash map.

Oh, I haven't found the root cause of the original case for the time being, but when I have repeatedly executed `testCloseConnectionOnBrokerRejectedRequest` many times, if don‘t clear`failureMap` will cause similar problems.

**Review by Jason918 — APPROVED** [review]
LGTM

**Review by Technoboy- — APPROVED** [review]

**Review by congbobo184 — APPROVED** [review]

**Pull request merged**